pkgs/code_assets/doc/linking_issues.md

```diff
@@ -0,0 +1,44 @@
+# Linking Issues with Native Assets
+
+This document describes common linking issues that can occur when using native
+assets in Dart and provides workarounds.
+
+## Conflicting with Embedder Library Dependencies
+
+When a native asset provides a library that is also a dependency of the embedder
+(e.g., a Flutter app), you may encounter runtime crashes. This is because the
+symbols from the already-loaded library can conflict with the symbols from the
+native asset. 
+
+This can happen, for example, when using a `package:sqlite3` hook in a Flutter
+application on Linux. The Flutter embedder already depends on `libsqlite3.so`,
+so when the Dart code tries to load the `sqlite3` native asset, the dynamic
+linker can get confused and resolve symbols to the wrong library, leading to a
+crash. 
+
+See [GitHub issue #2724](https://github.com/dart-lang/native/issues/2724) for more details.
+
+### Workaround 1: Symbol Prefixing
+
+Create a wrapper library that statically links the conflicting library (e.g.,
+SQLite) and re-exposes its symbols with a unique prefix. This avoids name
+collisions with the library loaded by the embedder.
+
+If your library allows the end-user to pass a dynamic library with user defines
+or use the system library, then the wrapper can be designed to to dynamically
+link against that library while still re-exposing the symbols with a prefix.
```

Comment:
Just my two cents: This isn't trivial to get right. Because if I do say

```C
int my_package_sqlite3_open_v2(
  const char *filename,
  sqlite3 **ppDb,
  int flags,
  const char *zVfs
) {
  return sqlite3_open_v2(filename, ppDb, flags, zVfs);
}
```

and use the default options from `native_toolchain_c` to link that and `sqlite3.c` into a shared library, my understanding is that `sqlite3_open_v2` from the previously loaded library takes precedence over the `sqlite3_open_v2` defined in the my wrapper library.

I think if I compile `sqlite3.c` with `-fvisibility=hidden` this could work, but then it's essentially the same thing as step 2 with extra steps.